### ISSUE TITLE
Conflict in latest versions of nuget packages.

### ISSUE THREAD

**Issue by SurferJeffAtGoogle** [GoogleCloudPlatform/google-cloud-dotnet#596]

**Comment by SurferJeffAtGoogle** [comment]
To reproduce:
Open https://github.com/SurferJeffAtGoogle/getting-started-dotnet/blob/issue596/aspnet/5-pubsub/5-pubsub.sln with Visual Studio.
Run all tests.

Error message:

```
Test Name:	GoogleCloudSamples.BookDetailLookupTest.TestLoop
Test FullName:	GoogleCloudSamples.BookDetailLookupTest.TestLoop
Test Source:	C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\aspnet\5-pubsub\test\BookDetailLookupTest.cs : line 118
Test Outcome:	Failed
Test Duration:	0:00:00.556

Result StackTrace:	
at Google.Api.Gax.Grpc.ChannelPool.GetChannel(ServiceEndpoint endpoint)
   at Google.Pubsub.V1.PublisherClient.Create(ServiceEndpoint endpoint, PublisherSettings settings)
   at GoogleCloudSamples.Services.BookDetailLookup..ctor(String projectId, Options options, ISimpleLogger logger) in C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\aspnet\5-pubsub\lib\Services\BookDetailLookup.cs:line 69
   at GoogleCloudSamples.BookDetailLookupTest.NewBookDetailLookup() in C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\aspnet\5-pubsub\test\BookDetailLookupTest.cs:line 87
   at GoogleCloudSamples.BookDetailLookupTest.TestLoop() in C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\aspnet\5-pubsub\test\BookDetailLookupTest.cs:line 119
Result Message:	System.InvalidOperationException : Error reading credential file from location C:\Users\Jeffrey Rennie\Documents\owner-bookshelf-dotnet.json: Could not load file or assembly 'BouncyCastle.Crypto, Version=1.7.4137.9688, Culture=neutral, PublicKeyToken=a4292a325f69b123' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
```

**Comment by jskeet** [comment]
Your `test` project has a nuget reference to BouncyCastle 1.8.1 ([here](https://github.com/SurferJeffAtGoogle/getting-started-dotnet/blob/issue596/aspnet/5-pubsub/test/packages.config#L3)). That isn't compatible with 1.7.0, and is causing the problem. This is due to the BouncyCastle team changing their signing key between versions.

In Q1 2017 we're hoping to create a new major version of the REST libraries with BouncyCastle 1.8.1, but until then you need to target 1.7.0 in your packages file.

**Comment by SurferJeffAtGoogle** [comment]
This broken state results from running `nuget update`.
I'm surprised that *any*  `nuget update` would break the code.  Why does it?
```
nuget update -Prerelease C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\aspnet\5-pubsub\5-pubsub.sln -Id Goog
le.Api.Gax -Id Google.Api.Gax.Rest -Id Google.Apis -Id Google.Apis.Auth -Id Google.Apis.Core -Id Google.Apis.Storage.v
1 -Id Google.Storage.V1 -Id Google.Api.CommonProtos -Id Google.Api.Gax -Id Google.Api.Gax.Grpc -Id Google.Apis -Id Goo
gle.Apis.Auth -Id Google.Apis.Core -Id Google.Datastore.V1 -Id Google.Iam.V1 -Id Google.Protobuf -Id Google.Pubsub.V1
Scanning for projects...
MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'.
Found 4 projects with a packages.config file. (bookshelf.csproj, lib.csproj, test.csproj, worker.csproj)
Feeds used:
  https://api.nuget.org/v3/index.json
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

Attempting to gather dependency information for multiple packages with respect to project 'C:\Users\Jeffrey Rennie\git
repos\getting-started-dotnet\aspnet\5-pubsub\bookshelf\bookshelf.csproj', targeting '.NETFramework,Version=v4.5'
Attempting to resolve dependencies for multiple packages.
Resolving actions install multiple packages
Removed package 'Google.Apis.Storage.v1 1.18.0.658' from 'packages.config'
Successfully uninstalled 'Google.Apis.Storage.v1 1.18.0.658' from bookshelf.csproj
Removed package 'Google.Apis.Auth 1.18.0' from 'packages.config'
Successfully uninstalled 'Google.Apis.Auth 1.18.0' from bookshelf.csproj
Removed package 'Google.Apis 1.18.0' from 'packages.config'
Successfully uninstalled 'Google.Apis 1.18.0' from bookshelf.csproj
Removed package 'Google.Apis.Core 1.18.0' from 'packages.config'
Successfully uninstalled 'Google.Apis.Core 1.18.0' from bookshelf.csproj
For adding package 'Google.Apis.Core.1.19.0' to project 'bookshelf.csproj' that targets 'net45'.
For adding package 'Google.Apis.Core.1.19.0' to project 'bookshelf.csproj' that targets 'net45'.
Package 'Google.Apis.Core.1.19.0' already exists in folder 'C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\as
pnet\5-pubsub\packages'
Added package 'Google.Apis.Core.1.19.0' to 'packages.config'
Successfully installed 'Google.Apis.Core 1.19.0' to bookshelf.csproj
For adding package 'Google.Apis.1.19.0' to project 'bookshelf.csproj' that targets 'net45'.
For adding package 'Google.Apis.1.19.0' to project 'bookshelf.csproj' that targets 'net45'.
Package 'Google.Apis.1.19.0' already exists in folder 'C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\aspnet\
5-pubsub\packages'
Added package 'Google.Apis.1.19.0' to 'packages.config'
Successfully installed 'Google.Apis 1.19.0' to bookshelf.csproj
For adding package 'Google.Apis.Auth.1.19.0' to project 'bookshelf.csproj' that targets 'net45'.
For adding package 'Google.Apis.Auth.1.19.0' to project 'bookshelf.csproj' that targets 'net45'.
Package 'Google.Apis.Auth.1.19.0' already exists in folder 'C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\as
pnet\5-pubsub\packages'
Added package 'Google.Apis.Auth.1.19.0' to 'packages.config'
Successfully installed 'Google.Apis.Auth 1.19.0' to bookshelf.csproj
For adding package 'Google.Apis.Storage.v1.1.19.0.678' to project 'bookshelf.csproj' that targets 'net45'.
For adding package 'Google.Apis.Storage.v1.1.19.0.678' to project 'bookshelf.csproj' that targets 'net45'.
Package 'Google.Apis.Storage.v1.1.19.0.678' already exists in folder 'C:\Users\Jeffrey Rennie\gitrepos\getting-started
-dotnet\aspnet\5-pubsub\packages'
Added package 'Google.Apis.Storage.v1.1.19.0.678' to 'packages.config'
Successfully installed 'Google.Apis.Storage.v1 1.19.0.678' to bookshelf.csproj
Feeds used:
  https://api.nuget.org/v3/index.json
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

Attempting to gather dependency information for multiple packages with respect to project 'C:\Users\Jeffrey Rennie\git
repos\getting-started-dotnet\aspnet\5-pubsub\lib\lib.csproj', targeting '.NETFramework,Version=v4.5'
Attempting to resolve dependencies for multiple packages.
Resolving actions install multiple packages
Removed package 'Google.Apis.Auth 1.18.0' from 'packages.config'
Successfully uninstalled 'Google.Apis.Auth 1.18.0' from lib.csproj
Removed package 'Google.Apis 1.18.0' from 'packages.config'
Successfully uninstalled 'Google.Apis 1.18.0' from lib.csproj
Removed package 'Google.Apis.Core 1.18.0' from 'packages.config'
Successfully uninstalled 'Google.Apis.Core 1.18.0' from lib.csproj
For adding package 'Google.Apis.Core.1.19.0' to project 'lib.csproj' that targets 'net45'.
For adding package 'Google.Apis.Core.1.19.0' to project 'lib.csproj' that targets 'net45'.
Package 'Google.Apis.Core.1.19.0' already exists in folder 'C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\as
pnet\5-pubsub\packages'
Added package 'Google.Apis.Core.1.19.0' to 'packages.config'
Successfully installed 'Google.Apis.Core 1.19.0' to lib.csproj
For adding package 'Google.Apis.1.19.0' to project 'lib.csproj' that targets 'net45'.
For adding package 'Google.Apis.1.19.0' to project 'lib.csproj' that targets 'net45'.
Package 'Google.Apis.1.19.0' already exists in folder 'C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\aspnet\
5-pubsub\packages'
Added package 'Google.Apis.1.19.0' to 'packages.config'
Successfully installed 'Google.Apis 1.19.0' to lib.csproj
For adding package 'Google.Apis.Auth.1.19.0' to project 'lib.csproj' that targets 'net45'.
For adding package 'Google.Apis.Auth.1.19.0' to project 'lib.csproj' that targets 'net45'.
Package 'Google.Apis.Auth.1.19.0' already exists in folder 'C:\Users\Jeffrey Rennie\gitrepos\getting-started-dotnet\as
pnet\5-pubsub\packages'
Added package 'Google.Apis.Auth.1.19.0' to 'packages.config'
Successfully installed 'Google.Apis.Auth 1.19.0' to lib.csproj
Feeds used:
  https://api.nuget.org/v3/index.json
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

Feeds used:
  https://api.nuget.org/v3/index.json
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
```

**Comment by SurferJeffAtGoogle** [comment]
The state of the project before running `nuget update` is here:
https://github.com/GoogleCloudPlatform/getting-started-dotnet/tree/master/aspnet/5-pubsub

**Comment by jskeet** [comment]
I don't know whether Nuget update would have updated to a new major version - I hope not. In this case it's a minor version bump which *should* be fine - but BouncyCastle has done the wrong thing, basically.

**Comment by SurferJeffAtGoogle** [comment]
The prior test/packages.config file has no reference to BouncyCastle at all.

**Comment by jskeet** [comment]
Interesting. There's not a lot we can do about this within this repo. There's a [PR](https://github.com/google/google-api-dotnet-client/pull/878/files) in the REST support library which would link BouncyCastle statically - I don't know whether that will go anywhere though, as I suspect there are some other issues to consider. If we could find another way of removing the BouncyCastle dependency, that would be ideal.

We'll keep giving this some thought - it's definitely not a pleasant situation.

// cc @chrisdunelm

**Comment by jskeet** [comment]
Fixed now, in the soon-to-be-released versions that target GAX 1.0.0-beta06, which in turn targets Google.Apis.Auth 1.0.20.
